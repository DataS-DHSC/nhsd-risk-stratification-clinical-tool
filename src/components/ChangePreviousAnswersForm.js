import React from 'react';
import PropTypes from 'prop-types';
import { ChevronLeftIcon } from 'nhsuk-react-components';
import HiddenStateHolder from './HiddenStateHolder';
import ClientSideRoutingForm from './ClientSideRoutingForm';
import useShallowRouting from './form/useShallowRouting';

const ChangePreviousAnswersForm = ({
  formBaseUrl,
  prevUrl,
  method,
  stateFields,
}) => {
  const buttonOnClick = useShallowRouting(formBaseUrl, prevUrl);

  return (
    <ClientSideRoutingForm
      action={`${formBaseUrl}/${prevUrl}`}
      method={method}
      id="change-previous-answers-form"
    >
      <HiddenStateHolder stateFields={stateFields} />

      <div className="nhsuk-back-link">
        <button
          type="submit"
          className="nhsuk-back-link__link"
          id="back-link"
          onClick={buttonOnClick}
        >
          <ChevronLeftIcon /> Change my previous answers
        </button>
      </div>
    </ClientSideRoutingForm>
  );
};

ChangePreviousAnswersForm.propTypes = {
  stateFields: PropTypes.shape().isRequired,
  method: PropTypes.string.isRequired,
  prevUrl: PropTypes.string.isRequired,
  formBaseUrl: PropTypes.string.isRequired,
};

export default ChangePreviousAnswersForm;
