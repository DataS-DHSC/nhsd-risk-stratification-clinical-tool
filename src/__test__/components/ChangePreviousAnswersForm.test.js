import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { useRouter } from 'next/router';
import ChangePreviousAnswersForm from '../../components/ChangePreviousAnswersForm';

jest.mock('next/router');

describe('ChangePreviousAnswersForm', () => {
  it('should match snapshot', () => {
    const router = { push: jest.fn() };
    useRouter.mockImplementationOnce(() => router);

    const props = {
      stateFields: { field1: 'value1', field2: 'value2' },
      formBaseUrl: 'baseUrl',
      prevUrl: 'prevUrl',
      method: 'POST',
    };

    const container = render(<ChangePreviousAnswersForm {...props} />);
    expect(container.asFragment()).toMatchSnapshot();

    fireEvent.click(screen.getByText(/Change my previous answers/i));

    expect(router.push).toHaveBeenCalledWith(
      { pathname: 'baseUrl/prevUrl' },
      undefined,
      { shallow: true }
    );
  });
});
