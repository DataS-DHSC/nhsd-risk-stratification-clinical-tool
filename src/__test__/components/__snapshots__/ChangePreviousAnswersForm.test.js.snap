// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePreviousAnswersForm should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="baseUrl/prevUrl"
    id="change-previous-answers-form"
    method="POST"
  >
    <input
      name="field1"
      type="hidden"
      value="value1"
    />
    <input
      name="field2"
      type="hidden"
      value="value2"
    />
    <div
      class="nhsuk-back-link"
    >
      <button
        class="nhsuk-back-link__link"
        id="back-link"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__chevron-left"
          height="32"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
          />
        </svg>
         Change my previous answers
      </button>
    </div>
  </form>
</DocumentFragment>
`;
